tests/test_components.py

```diff
@@ -481,37 +541,80 @@ def test_parse(self):
 
 
 class CollectionsTestCase(TestCase):
-    def test_not_enabled(self):
-        component = Collections(Mock())
-        component.raw_data = {"enable": False}
-        component.parse()
-        self.assertFalse(ModerationCollection.objects.exists())
-
-    def test_enabled(self):
-        user1 = UserFactory()
-        user2 = UserFactory()
-        role1 = Role.objects.create(name="Role 1", user=user1)
-        role2 = Role.objects.create(name="Role 2", user=user2)
-        versions = PageVersionFactory.create_batch(8)
-        pages = [version.content.page for version in versions]
-        workflow = Workflow.objects.create(name="Workflow 1", is_default=False)
-        workflow.steps.create(role=role1, order=1)
-        workflow.steps.create(role=role2, order=2)
-        bootstrap = Mock(
-            users={"moderator": user1, "moderator2": user1, "moderator3": user1},
-            pages={
-                "page1": pages[0],
-                "page2": pages[1],
-                "page3": pages[2],
-                "page4": pages[3],
-                "page5": pages[4],
-                "page6": pages[5],
-                "page7": pages[6],
-                "page8": pages[7],
-            },
-            workflows={"wf4": workflow, "wf5": workflow},
-        )
-        component = Collections(bootstrap)
-        component.raw_data = {"enable": True}
+    def setUp(self):
+        self.user = UserFactory(username="user1")
+        self.wf1 = Workflow.objects.create(name="Workflow 1")
+        self.version = PageVersionFactory()
+        self.page1 = self.version.content.page
+        self.bootstrap = Mock(users={"user1": self.user}, workflows={"wf1": self.wf1}, pages={"page1": self.page1})
+        self.component = Collections(self.bootstrap)
+
+    def test_parse(self):
+        """
+        Test that parsing is able to call the Collection.get_or_create method
+        """
+        component = self.component
+        component.raw_data = {'collection1': {'pages': ['page1'], 'name': 'Collection 1', 'user': 'user1', 'workflow': 'wf1'}}            
+
+        with patch.object(
+            component, 'get_version', return_value=self.version
+        ) as get_version, patch.object(
+            component, 'get_or_create'
+        ) as get_or_create, patch.object(
+            component, 'add_version'
+        ) as add_version:
+            component.parse()
+            get_or_create.assert_called_once_with({'workflow': self.wf1, 'name': 'Collection 1', 'author': self.user})
```

Comment:
So tbh, this test is a maintainability nightmare. I've worked on codebases with tests like these and it's a horrible experience. This test will not break if `parse` ever stops doing what it expected of it (taking a dict of values and saving it to db or do nothing if it's already in the db), but it will break if anybody tries to refactor `parse` (like say do what me and Riz have both suggested in this PR). This is a terrible use of unit testing, it literally just makes things more difficult without adding anything of value.